vm/src/obj/objsequence.rs

```diff
@@ -86,8 +85,20 @@ pub trait PySliceableSequence {
                 } else {
                     // calculate the range for the reverse slice, first the bounds needs to be made
                     // exclusive around stop, the lower number
-                    let start = start.as_ref().map(|x| x + 1);
-                    let stop = stop.as_ref().map(|x| x + 1);
+                    let start = start.as_ref().map(|x| {
+                        if *x == (-1).to_bigint().unwrap() {
+                            self.len() + BigInt::one() //.to_bigint().unwrap()
+                        } else {
+                            x + 1
+                        }
+                    });
+                    let stop = stop.as_ref().map(|x| {
+                        if *x == (-1).to_bigint().unwrap() {
```

Comment:
Does it make sense to have a special case for -1? I do not understand what is happening here.

Reply:
I'll try to explain but it's not easy. look at it , lhs is what python  is doing and rhs is the way we are trying to do it (without fix).
```python
In [26]: "abcd"[:-1:-1], "abcd"[0:][::-1] # stop without fix                    
Out[26]: ('', 'dcba')

In [27]: "abcd"[:-1:-1], "abcd"[5:][::-1] #stop with fix                        
Out[27]: ('', '')

In [28]: "abcd"[-1::-1], "abcd"[:0][::-1] # start without fix                   
Out[28]: ('dcba', '')

In [29]: "abcd"[-1::-1], "abcd"[:4+1][::-1] # start with fix                    
Out[29]: ('dcba', 'dcba')
```
```python
In [43]: for i in range(-5,6): 
    ...:     print(i, "|",   'abcd'[:i:-1], "|", "abcd"[i+1:])                  
-5 | dcba | abcd
-4 | dcb | bcd
-3 | dc | cd
-2 | d | d
-1 |  | abcd
0 | dcb | bcd
1 | dc | cd
2 | d | d
3 |  | 
4 |  | 
5 |  | 

In [44]: for i in range(-5,6): 
    ...:     print(i, "|",   'abcd'[i::-1], "|", "abcd"[:i+1])                  
-5 |  | 
-4 | a | a
-3 | ba | ab
-2 | cba | abc
-1 | dcba | 
0 | a | a
1 | ba | ab
2 | cba | abc
3 | dcba | abcd
4 | dcba | abcd
5 | dcba | abcd
```

look at those example maybe it'll help to understand

Reply:
i would add that about slice, None is not the same thing as 0. and if think BigInt turns None to 0 ?

Reply:
Thank you for your detailed explanation, still, I find this hard to understand.

Did you test with a stop value of `-2`?

Reply:
yes the added test, tests every possible combination from
 start = range(-7, 7) to   end = range(-7, 7) with   step =range(-5, 5) for sequence of len = 4